app/test/gas_estimation_test.go

```diff
@@ -0,0 +1,191 @@
+package app
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/v3/app"
+	"github.com/celestiaorg/celestia-app/v3/app/encoding"
+	"github.com/celestiaorg/celestia-app/v3/app/grpc/gasestimation"
+	"github.com/celestiaorg/celestia-app/v3/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/v3/pkg/user"
+	"github.com/celestiaorg/celestia-app/v3/test/util/blobfactory"
+	"github.com/celestiaorg/celestia-app/v3/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/v3/test/util/testnode"
+	blobtypes "github.com/celestiaorg/celestia-app/v3/x/blob/types"
+	"github.com/celestiaorg/go-square/v2/share"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+func TestEstimateGasPrice(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping app/test/gas_estimation gas price and usage in short mode.")
+	}
+
+	// test setup: create a test chain, submit a few PFBs to it, keep track of their gas
+	// price, then test the gas estimator API.
+	accountNames := testfactory.GenerateAccounts(150) // using 150 to have 2 pages of txs
+	cfg := testnode.DefaultConfig().WithFundedAccounts(accountNames...).
+		WithTimeoutCommit(10 * time.Second) // to have all the transactions in just a few blocks
+	cctx, _, _ := testnode.NewNetwork(t, cfg)
+	require.NoError(t, cctx.WaitForNextBlock())
+
+	// create the gas estimation client
+	gasEstimationAPI := gasestimation.NewGasEstimatorClient(cctx.GRPCClient)
```

Comment:
these could be different follow upPRs, or addressed here up to you:
- add simple helper function/method in the signer that creates this client, queries for the gas price, and returns it 
- add docs somewhere for this client or that method/function

Reply:
sounds good 👍 I'll open an issue and do it once this is merged.

Where do you suggest this to be documented?

Reply:
https://github.com/celestiaorg/celestia-app/issues/4258

Reply:
probably wherever the signer docs are